src/thread.rs

```diff
@@ -603,6 +636,63 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         id
     }
 
+    #[inline]
+    fn start_thread(
+        &mut self,
+        thread: Option<&OpTy<'tcx, Tag>>,
+        start_routine: &OpTy<'tcx, Tag>,
+        start_abi: Abi,
+        arg: &OpTy<'tcx, Tag>,
```

Comment:
Looking at the other APIs in this function, I think we shouldn't use `OpTy` here, but rather something more specific -- `Scalar<Tag>` or `Pointer<Option<Tag>>` or so.